BasicSample/Assets/GrabAndThrow/Scripts/GrabAndThrowManager.cs

```diff
@@ -0,0 +1,143 @@
+		}
+
+		// Information describing the state of one of the hands
+		private class HandData
+		{
+			public bool wasTapped = false;
+			public Transform transform;
+			public ObjectHeldData objectHeld = null;
+		}
+
+		private HandData m_leftHandData = new HandData();
+		private HandData m_rightHandData = new HandData();
+		private ThrowableObject[] m_throwableObjects;
+
+		private void Awake()
+		{
+			// Create two empty GameObjects for their Transforms. These Transforms are useful for calculating and
+			// maintaining offsets between the hands and the objects when grabbing and releasing them.
+			GameObject leftHandSpaceGameObject = new GameObject("Left Hand Space");
+			m_leftHandData.transform = Instantiate(leftHandSpaceGameObject, Vector3.zero, Quaternion.identity).transform;
+			GameObject rightHandSpaceGameObject = new GameObject("Right Hand Space");
+			m_rightHandData.transform = Instantiate(rightHandSpaceGameObject, Vector3.zero, Quaternion.identity).transform;
+
+			// Find potential objects to pick up, searching the children of this gameObject.
+			m_throwableObjects = GetComponentsInChildren<ThrowableObject>();
+		}
+
+		void Update()
+		{
+			UpdateForHand(XRNode.LeftHand, m_leftHandData);
+			UpdateForHand(XRNode.RightHand, m_rightHandData);
+		}
+
+		// For each given hand: Grab, Move, or Release a GameObject
+		void UpdateForHand(XRNode handNode, HandData handData)
+		{
+			InputDevice device = InputDevices.GetDeviceAtXRNode(handNode);
+			bool deviceHasData = device.TryGetFeatureValue(CommonUsages.isTracked, out bool deviceIsTracked);
+			deviceHasData &= device.TryGetFeatureValue(CommonUsages.primaryButton, out bool isDeviceTapped);
+			deviceHasData &= device.TryGetFeatureValue(CommonUsages.devicePosition, out Vector3 devicePosition);
+			deviceHasData &= device.TryGetFeatureValue(CommonUsages.deviceRotation, out Quaternion deviceRotation);
+
+			if (!deviceHasData || !deviceIsTracked)
+			{
+				return;
+			}
+
+			// Update the hand's transform for use below
+			handData.transform.SetPositionAndRotation(devicePosition, deviceRotation);
+
+			// If this hand should grab an object, look for one nearby
+			if (handData.objectHeld == null && isDeviceTapped && !handData.wasTapped)
+			{
+				ThrowableObject nearestObject = FindNearestObject(m_throwableObjects, devicePosition);
+				if (nearestObject != null)
+				{
+					// Grab the object
+					handData.objectHeld = new ObjectHeldData { rigidbody = nearestObject.GetComponent<Rigidbody>() };
+					handData.objectHeld.rigidbody.isKinematic = true;
+
+					// Record the offset from the hand pose to the object origin pose
+					handData.objectHeld.positionInHandSpace = handData.transform.InverseTransformPoint(handData.objectHeld.rigidbody.position);
+					handData.objectHeld.rotationInHandSpace = Quaternion.Inverse(deviceRotation) * handData.objectHeld.rigidbody.rotation;
+				}
+			}
+
+			// If this hand is holding an object, update its position 
```

Comment:
super minor.  "update its position and rotation" or "pose".

Reply:
Thanks! It's "pose" now in the latest change.